src/python/pants/backend/python/util_rules/pex_cli.py

```diff
@@ -42,6 +43,10 @@ class PexCli(TemplatedExternalTool):
     default_url_template = "https://github.com/pex-tool/pex/releases/download/{version}/pex"
     version_constraints = ">=2.3.0,<3.0"
 
+    # extra args to be passed to the pex tool; note that they
+    # are going to apply to all invocations of the pex tool.
+    global_args = ArgsListOption(example="--check=error --no-compile")
```

Comment:
I realised that this comment won't be visible to any users, maybe we could pass it as `extra_help`, which is added to the end of the main doc string. Might also add some examples of the variety of places that pex is invoked:

```suggestion
    global_args = ArgsListOption(
        example="--check=error --no-compile", 
        extra_help="Note that these apply to all invocations of the pex tool, including building `pex_binary` targets, preparing `python_test` targets to run, and generating lockfiles."
    )
```

Or something like that? We could potentially even suggest that this might typically be used as a CLI flag rather than putting it into a toml file, but that feels a bit unnecessary 🤷‍♂️

Reply:
I like this a lot. Yes, I think mentioning cli flags would be unnecessary - the help is clear about this. I think this is going to be a very niche usage, but I think also a very powerful one as it unties you from having support for a particular pex version in Pants.

I've used your docs with a softwrap.